backend/src/common/client/api_server/v2/experiment_client.go

```diff
@@ -0,0 +1,217 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api_server_v2
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	apiclient "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/experiment_client"
+	params "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/experiment_client/experiment_service"
+	model "github.com/kubeflow/pipelines/backend/api/v2beta1/go_http_client/experiment_model"
+	"github.com/kubeflow/pipelines/backend/src/common/client/api_server"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	"golang.org/x/net/context"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	apiServerDefaultTimeout = 35 * time.Second
+)
+
+type ExperimentInterface interface {
+	Create(params *params.CreateExperimentParams) (*model.V2beta1Experiment, error)
+	Get(params *params.GetExperimentParams) (*model.V2beta1Experiment, error)
+	List(params *params.ListExperimentsParams) ([]*model.V2beta1Experiment, int, string, error)
+	ListAll(params *params.ListExperimentsParams, maxResultSize int) ([]*model.V2beta1Experiment, error)
+	Archive(params *params.ArchiveExperimentParams) error
+	Unarchive(params *params.UnarchiveExperimentParams) error
+}
+
+type ExperimentClient struct {
+	apiClient      *apiclient.Experiment
+	authInfoWriter runtime.ClientAuthInfoWriter
+}
+
+func NewExperimentClient(clientConfig clientcmd.ClientConfig, debug bool) (
+	*ExperimentClient, error) {
+
+	runtime, err := api_server.NewHTTPRuntime(clientConfig, debug)
+	if err != nil {
+		return nil, fmt.Errorf("Error occurred when creating experiment client: %w", err)
+	}
+
+	apiClient := apiclient.New(runtime, strfmt.Default)
+
+	// Creating experiment client
+	return &ExperimentClient{
+		apiClient:      apiClient,
+		authInfoWriter: api_server.PassThroughAuth,
+	}, nil
+}
+
+func NewKubeflowInClusterExperimentClient(namespace string, debug bool) (
+	*ExperimentClient, error) {
+
+	runtime := api_server.NewKubeflowInClusterHTTPRuntime(namespace, debug)
+
+	apiClient := apiclient.New(runtime, strfmt.Default)
+
+	// Creating experiment client
+	return &ExperimentClient{
+		apiClient:      apiClient,
+		authInfoWriter: api_server.SATokenVolumeProjectionAuth,
+	}, nil
+}
+
+func (c *ExperimentClient) Create(parameters *params.CreateExperimentParams) (*model.V2beta1Experiment,
+	error) {
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), apiServerDefaultTimeout)
+	defer cancel()
+
+	// Make service call
+	parameters.Context = ctx
+	response, err := c.apiClient.ExperimentService.CreateExperiment(parameters)
+	if err != nil {
+		return nil, util.NewUserError(err,
+			fmt.Sprintf("Failed to create experiment. Params: '%+v'. Body: '%+v'", parameters, parameters.Body),
+			fmt.Sprintf("Failed to create experiment '%v'", parameters.Body.DisplayName))
```

Comment:
NIT: combine these as `fmt.Sprintf("Failed to create experiment '%v'. Params: '%+v'. Body: '%+v'", parameters.Body.DisplayName, parameters, parameters.Body)`